src/rpc/chain_api.rs

```diff
@@ -29,12 +30,17 @@ pub(in crate::rpc) async fn chain_get_message<DB: Blockstore>(
     data: Data<RPCState<DB>>,
     Params(LotusJson((msg_cid,))): Params<LotusJson<(Cid,)>>,
 ) -> Result<LotusJson<Message>, JsonRpcError> {
-    let ret: Message = data
-        .state_manager
-        .blockstore()
-        .get_cbor(&msg_cid)?
-        .ok_or("can't find message with that cid")?;
-    Ok(LotusJson(ret))
+    let bytes = data.state_manager.blockstore().get(&msg_cid)?;
+    match bytes {
+        Some(bz) => match fvm_ipld_encoding::from_slice::<Message>(&bz) {
```

Comment:
what is `bz`?

Reply:
Hmm not sure, let me rename it, it's from `fvm_ipld_encoding` code
```rust
/// Get typed object from block store by Cid.
    fn get_cbor<T>(&self, cid: &Cid) -> anyhow::Result<Option<T>>
    where
        T: de::DeserializeOwned,
    {
        match self.get(cid)? {
            Some(bz) => {
                let res = crate::from_slice(&bz)?;
                Ok(Some(res))
            }
            None => Ok(None),
        }
    }

```

Reply:
Fixed by using `enum ChainMessage` instead